corehq/apps/users/models.py

```diff
@@ -562,13 +562,10 @@ def transfer_domain_membership(self, domain, to_user, create_record=False, is_ad
 
     @memoized
     def is_domain_admin(self, domain=None):
+        # this is a hack needed because we can't pass parameters from views
+        domain = domain or getattr(self, 'current_domain', None)
         if not domain:
-            # hack for template
-            if hasattr(self, 'current_domain'):
-                # this is a hack needed because we can't pass parameters from views
-                domain = self.current_domain
-            else:
-                return False  # no domain, no admin
```

Comment:
this block could still end up with a null domain if `self.current_domain = None`

Reply:
Writing this out because I'm having difficulty following.  When this is called without a `domain` arg there are two big forks - whether or not this was called inside a normal request, where `current_domain` would be set, possibly to `None` if it's not a domain page.

For a global admin, the previous behavior was:
If `current_domain` is set, always allow, as long as this isn't known to be a domain which restricts superusers.  If it's not set, always block.

The new behavior is:
If `current_domain` is set, allow as long as the domain is known and doesn't restrict superusers.  Otherwise block.

For a non-global admin, I don't think there's a change (assuming `get_domain_membership(None)` returns `None`)

That sounds like a fairly safe change - I think the failure mode would be that superusers could in theory be denied permission if accessing something that can't find the domain, whereas they couldn't before.

Reply:
That's correct. This change handles an empty domain the same as None (return False)

> whether or not this was called inside a normal request, where current_domain would be set, possibly to None if it's not a domain page.

`user.current_domain` should never be set to None: https://github.com/dimagi/commcare-hq/search?q=%22user.current_domain+%3D%22&type=